src/mbgl/map/geometry_tile.cpp

```diff
@@ -0,0 +1,16 @@
+#include <mbgl/map/geometry_tile.hpp>
+#include <mbgl/style/filter_expression_private.hpp>
+
+namespace mbgl {
+
+mapbox::util::optional<Value> GeometryTileFeatureExtractor::getValue(const std::string& key) const {
+    if (key == "$type") {
+        return Value(uint64_t(feature.getType()));
+    }
+
+    return feature.getValue(key);
+}
+
+template bool evaluate(const FilterExpression&, const GeometryTileFeatureExtractor&);
```

Comment:
Can we make this a non-free function or rename it some way so that we don't have a "evaluate" function in the global mbgl namespace?